File: striver_sheet/coding-solution/largest_subarray_with_0_sum.cpp
class Solution{
    public:
    int maxLen(vector<int>&A, int n)
    {   
        unordered_map<int, int> prefixSum;
        int idx = -1;
        int max_count = 0;
        
        int sum = 0;
        for(int i=0; i<A.size(); i++){
            sum += A[i];
            int count = 0;
            if(sum==0){
                max_count = i+1;
            }
            else{
                if(prefixSum.find(sum)!=prefixSum.end()){
                    count = i - prefixSum[sum];
                    max_count = max(max_count, count);
                 
                }
                else{
                    prefixSum.insert({sum, i});
                }   
            }
        }
        
        return max_count;
    }
};

File: python-daily/largest_subarray_with_0_sum.py
class Solution:
    def maxLen(self, n, arr):
        #Code here
        max_count = 0
        prefix_sum = 0
        arr_dict = dict()
        
        for i in range(n):
            prefix_sum += arr[i]
            
            if prefix_sum==0:
                max_count = max(max_count, i+1)
            
            elif prefix_sum in arr_dict:
                max_count = max(max_count, i-arr_dict[prefix_sum])
            else:
                arr_dict[prefix_sum] = i
            
        return max_count